xsuite/src/world/signer.ts

```diff
@@ -56,26 +56,35 @@ export class KeystoreSigner extends UserSigner {
 export class Keystore {
   data: any;
   password: string;
+  kind: "mnemonic" | "secretKey";
 
   constructor(data: any, password: string) {
+    if (!["mnemonic", "secretKey", undefined].includes(data.kind)) {
+      throw new Error(
+        `Invalid kind value: "${data.kind}". It must be "mnemonic", "secretKey", or undefined.`,
+      );
+    }
```

Comment:
Tell me if you want to catch this somewhere.

Reply:
Would be great to have a helper `getSupportedKeystoreKind` that raises the error and return the kind, including the default value in case it is not specified.

Reply:
👍